### ISSUE TITLE
It should feature a callback to next location

### ISSUE THREAD

**Issue by karlgroves** [svinkle/launchy#8]
Not sure if related to issue #6 or not, but one of the things I've realized when creating features like this is that the "next location" needs to be set. You're already doing this when `esc` is hit or dialog is closed, but there may be cases where the next focused location is something else (based on user interaction). So having an explicit place to send users is a good feature

**Comment by svinkle** [comment]
@karlgroves Hmm, interesting. I could have another optional `data-launchy-next` type attribute where the value would the the `id` of the element to set focus to on close.

I've not heard of this as a usability requirement before. What type of context might such a thing be used with?

**Comment by karlgroves** [comment]
In terms of use case: this feature would be necessary whenever the next logical element is not the element that invoked the dialog.

In other words:
* We assume the dialog is opened on purpose, but that may not be the case. The user might have opened it on accident, thinking it does something else. Or they may open it and change their mind. Or there might be some other reason that's not what the user wants. For this, you support striking the `esc` or an explicit 'Close' affordance.
* Then there's the case where the user did launch the dialog, did interact with it in some meaningful way, and there's a business reason for them to go somewhere else. Imagine a case, where the dialog is used as a way to  "Yes" or "No" options. Answering "Yes" does something and requires focus to go to position `x` on the screen or answering "No" does something entirely different and requires focus to go to position `y`.  Hence the need to track the next location of focus.

I encountered this in the real world not too long ago where a client launched dialogs from other dialogs depending on the workflow. It was a really terrible UI overall, but we could at least mitigate some of the a11y issues by ensuring that focus was explicitly set to the right place based on other logic that has occurred elsewhere - hence the callback.

**Comment by svinkle** [comment]
@karlgroves Thanks for the input! Interesting stuff. 

I'm thinking for the case where button "Yes" goes to `x` and "No" goes to `y`; that would be up to the developer implementing these buttons to shift focus on click? All I'd need to do is expose the method to close the dialog window for them to add to their click event handler. I'm not sure if this needs to be a feature add? 🤔

It's just my immediate thought; maybe I'll have another think later on. I definitely need to add some way to handle custom use cases like this, though.

**Comment by svinkle** [comment]
@karlgroves Thoughts on the usability of the newly released feature per [v0.8.0](https://github.com/svinkle/launchy/releases/tag/v0.8.0)?

---

You can add your own custom "refocus" controls to any Launchy! modal window!

1. Add the HTML element you wish to use for the control within your content
2. Apply the `data-launchy-refocus` attribute
3. Set the value of the `data-launchy-refocus` attribute to the` id` of the element you wish to send focus to

Example:

```html
<div data-launchy data-launchy-text="Launch window!" data-launchy-title="My modal window">
    <p>This content will appear in the modal window.</p>
    <a href="#" data-launchy-refocus="post-title">Send me to the article headline!</a>
</div>

<!-- Somewhere else in your page… -->
<h1 id="post-title" tabindex="-1">My perfect sundae…</h1>
```

Clicking on the _"Send me to the article headline!"_ link will hide the modal window and shift keyboard focus to the `<h1 id="post-title"/>` element!

**Comment by karlgroves** [comment]
I like it!
